Common/Python/PandasData.cs

```diff
@@ -276,7 +277,14 @@ class Remapper(wrapt.ObjectProxy):
                         throw new ArgumentException($"PandasData.ctor(): More than one \'{duplicateKey}\' member was found in \'{type.FullName}\' class.");
                     }
 
-                    keys = members.Select(x => x.Name.ToLowerInvariant()).Except(_baseDataProperties).Concat(new[] { "value" });
+                    // If the custom data derives from a Market Data (e.g. Tick, TradeBar, QuoteBar), exclude its keys
+                    var baseType = type.BaseType;
+                    var inheritedProperties = baseType != null && baseType.Namespace == dataMarketNamespace
+                        ? baseType.GetProperties().ToHashSet(x => x.Name.ToLowerInvariant())
+                        : _baseDataProperties;
+
+                    keys = members.Select(x => x.Name.ToLowerInvariant()).Except(inheritedProperties).Concat(new[] {"value"});
```

Comment:
`members` is being re enumerated multiple times we probably want to convert it to a list?

If the issue is duplicated `keys` why not make it a set directly? 🤔 After a quick test I think its `columns` that ends up holding duplicates in `master` and causes the exception?

Reply:
What caused the exception was the fact that `_members` had `TradeBar` members. In this block [PandasData.cs#L296:L302](https://github.com/QuantConnect/Lean/blob/master/Common/Python/PandasData.cs#L296:L302), we were updating "open", "high", "low", "close" and "volume". After that, in [PandasData.cs#L319](https://github.com/QuantConnect/Lean/blob/master/Common/Python/PandasData.cs#L319), `tradeBar` would not be `null` since CBOE inherits from `TradeBar` and we would update "open", "high", "low", "close" and "volume" again.
So the idea is to remove the `TradeBar` properties from `_members`.

Reply:
Thanks for the explanation. The current solution wouldn't cover another level on inheritance, for example, a custom type inheriting from CBOE (it's a good unit test to add) since we are relying on the base type and its namespace.
Since this is just for `QuoteBar/TradeBar/Tick` https://github.com/QuantConnect/Lean/blob/master/Common/Python/PandasData.cs#L319 maybe the solution could be aware of it too - Simple one could be casting to `QuoteBar/TradeBar/Tick` (as done later) and checking if not null and removing those members 🤔 

While reviewing I realized that we are enumerating all the data, not sure the reason https://github.com/QuantConnect/Lean/pull/3748/files#diff-92c2966a841a7cd2c63b5a1ae6c9bffaR239-R246